### ISSUE TITLE
How can `ThreadPoolBuilder::build` fail?

### ISSUE THREAD

**Issue by ExpHP** [rayon-rs/rayon#579]
`ThreadPoolBuilder::build` returns a `Result<ThreadPool, ThreadPoolBuildError>`, which the example in the documentation [unceremoniously unwraps](https://docs.rs/rayon/1.0.1/rayon/struct.ThreadPoolBuilder.html#creating-a-threadpool).

After digging into the source, it seems that the error comes from [`std::thread::Builder::spawn`](https://doc.rust-lang.org/std/thread/struct.Builder.html#method.spawn), i.e. failure to create a thread at the OS level.  Not even `std::thread::spawn` bothers to return a Result for this, so I guess the unwrap is because this is such an absurdly uncommon scenario?

**Comment by cuviper** [comment]
Yes, the only errors are `GlobalPoolAlreadyInitialized` for `build_global` in particular, and OS errors creating the threads in general.  I don't know if I would call them *absurdly* uncommon, but that's subjective.  For example, you can easily hit this by requesting more threads than the current ulimit allows.

I'd be OK with changing example code to something more friendly, perhaps just `expect` instead.
